### PR TITLE
remove unnecessary index in CoinStore & add additional benchmarks 

### PR DESCRIPTION
**Pull request by arvidn**
this extends the benchmark a bit and improves performance by deleting an unused index in the sqlite coin store table.

As far as I can tell there are no queries that requires an index on whether a coin is spent or not. In fact, the mere existence of this index made the query planner mistakenly use it, causing a significant performance issue (as demonstrated by the benchmark).

My main attempt with this patch was to improve insertion performance, by removing the index and thus reducing the amount of work for insertions. This had a small effect in insertion performance, but more noticeable impact on the lookup speed of coins by name, excluding spent coins. The problem was (presumably) that sqlite chose to use the `coin_spent`  index *first*, followed by looking up the coin ID. By removing the index, the query instead looks up the coin ID first, and then discards it if the coin is spent, this is significantly faster.

I ran this benchmark on a fast machine with an SSD (Apple M1) as well as an older machine with spinning disk (Ubuntu amd64).

# summary

* `get_records_by_name()` filtering spent coins improved from 49s -> 2s on SSD and from 70s -> 1s on HDD. This is most likely due to the query planner using the wrong index.
* On SSD the additions timings improved somewhat. from 35s -> 30s

# SSD

Before:

```
MOSTLY ADDITIONS, time: 35.6544s additions: 400400 removals: 20000
MOSTLY REMOVALS, time: 8.8911s additions: 600 removals: 140000
FULLBLOCKS, time: 69.9725s additions: 400400 removals: 400000
GET RECORDS BY NAMES with spent, time: 0.7952s 200 lookups found 40000 coins in total
GET RECORDS BY NAMES without spent, time: 49.0541s 200 lookups found 20747 coins in total
GET COINS REMOVED AT HIGHT, time: 15.5742s 800 blocks, found 580000 coins in total
```

After:

```
MOSTLY ADDITIONS, time: 29.9418s additions: 400400 removals: 20000
MOSTLY REMOVALS, time: 10.9705s additions: 600 removals: 140000
FULLBLOCKS, time: 54.4167s additions: 400400 removals: 400000
GET RECORDS BY NAMES with spent, time: 0.9823s 200 lookups found 40000 coins in total
GET RECORDS BY NAMES without spent, time: 2.0775s 200 lookups found 20773 coins in total
GET COINS REMOVED AT HIGHT, time: 15.3979s 800 blocks, found 580000 coins in total
```

# Spinning disk

Before:

```
MOSTLY ADDITIONS, time: 51.8249s additions: 400400 removals: 20000
MOSTLY REMOVALS, time: 6.6823s additions: 600 removals: 140000
FULLBLOCKS, time: 76.4455s additions: 400400 removals: 400000
GET RECORDS BY NAMES with spent, time: 1.2238s 200 lookups found 40000 coins in total
GET RECORDS BY NAMES without spent, time: 70.1104s 200 lookups found 20730 coins in total
GET COINS REMOVED AT HIGHT, time: 22.4741s 800 blocks, found 580000 coins in total
```

After:

```
MOSTLY ADDITIONS, time: 51.6045s additions: 400400 removals: 20000
MOSTLY REMOVALS, time: 4.5481s additions: 600 removals: 140000
FULLBLOCKS, time: 71.1832s additions: 400400 removals: 400000
GET RECORDS BY NAMES with spent, time: 1.2226s 200 lookups found 40000 coins in total
GET RECORDS BY NAMES without spent, time: 0.9238s 200 lookups found 20732 coins in total
GET COINS REMOVED AT HEIGHT, time: 22.4992s 800 blocks, found 580000 coins in total
```

### CODE REVIEW

File: chia/full_node/coin_store.py [hunk 1]
```diff
@@ -55,7 +55,9 @@ async def create(cls, db_wrapper: DBWrapper, cache_size: uint32 = uint32(60000))
 
         await self.coin_record_db.execute("CREATE INDEX IF NOT EXISTS coin_spent_index on coin_record(spent_index)")
 
-        await self.coin_record_db.execute("CREATE INDEX IF NOT EXISTS coin_spent on coin_record(spent)")
+        # earlier versions of chia created this index despite no lookups needing
+        # it
+        await self.coin_record_db.execute("DROP INDEX IF EXISTS coin_spent")
```
Comment (arvidn) [inline chia/full_node/coin_store.py:60]: this is the relevant change

**Comment by mariano54** [comment]
I originally thought the index was necessary for querying `spent=0`, but it's not. This index has just caused performance issues, and it's good to remove it.

**Comment by emlowe** [comment]
Agreed - it's caused grief before.

**Review by Yostra — DISMISSED** [review]

**Comment by arvidn** [comment]
One thing to keep in mind is that if someone downgraded chia from this patch, they'll get an unpleasant surprise on startup.

It might be more conservative to first remove the statement that creates the index, and then a few versions later actually remove it.

**Comment by emlowe** [comment]
I think Bill and I were more comfortable with removing the create index command for this and then some subsequent release dropping the index. As mentioned this helps to avoid the aforementioned unpleasantness.

**Comment by arvidn** [comment]
I will update the PR

**Review by emlowe — APPROVED** [review]
ok

**Pull request merged**